### PR TITLE
Hotfix/collateral test

### PR DESCRIPTION
**Pull request by cjkoepke**
Fixes an edge case where a selected UTxO in `prepareCollateral` ends up being too small to cover the required collateral when many native assets are present.

### CODE REVIEW

File: packages/blaze-tx/src/TxBuilder.ts [hunk 1]
```diff
@@ -1607,14 +1608,25 @@ export class TxBuilder {
         this.body.setCollateral(collateralSet);
       }
 
-      this.body.setCollateralReturn(
```
Comment (Quantumplation) [inline packages/blaze-tx/src/TxBuilder.ts:1610]: I think it's probably better to:
- split checkAndAlterOutput into two sub functions, one which sets coin to the minUTxO, and one which checks that the UTxO is valid
- modify the coin selection so it will never select a UTxO that doesn't have `minUTxO(value - collateral) + collateral` ADA
- call *only* the `checkOutput` function here

Comment (cjkoepke) [reply packages/blaze-tx/src/TxBuilder.ts:1610]: Yeah that seems cleaner, I'll update.

Comment (cjkoepke) [reply packages/blaze-tx/src/TxBuilder.ts:1610]: Updated: https://github.com/butaneprotocol/blaze-cardano/pull/255/commits/f2ad0962be4041994505116d520f00808b35162d

**Review by Quantumplation — APPROVED** [review]
